lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -53,16 +56,17 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
               <.secret_with_badge
                 :for={{secret_name, _} <- Enum.sort(@secrets)}
                 secret_name={secret_name}
-                secret_origin={:session}
+                secret_hub_id={:session}
                 stored="Session"
                 active={secret_name == @prefill_secret_name}
                 target={@myself}
               />
               <.secret_with_badge
                 :for={secret <- @saved_secrets}
+                :if={!(secret.name in Map.keys(@secrets) and @secrets[secret.name] != secret.value)}
```

Comment:
```suggestion
                :if={!(is_map_key(@secrets, secret.name) and @secrets[secret.name] != secret.value)}
```

Reply:
But this may also work:

```suggestion
                :if={{secret.name, secret.value} in @secrets}
```

I am not sure. Please check :)